delivery/util.py

```diff
@@ -74,3 +78,34 @@ def update_available(
         return False
 
     return branch_info.greatest_version > awesomeversion.AwesomeVersion(os_id.VERSION_ID)
+
+
+def artefact_metadata_for_resource_node(
+    resource_node: cnudie.iter.ResourceNode,
+    type: str,
+    delivery_client: delivery.client.DeliveryServiceClient = None,
+) -> typing.Iterable[delivery.model.ArtefactMetadata]:
+    '''Return an iterable that contains all stored `ArtefactMetadata` of the given type for the
+    given resource node.
+
+    For possible values for `type` see `dso.model.Datatype`.
+    '''
+    if not delivery_client:
+        if not (delivery_client := ccc.delivery.default_client_if_available()):
+            raise ValueError('No default delivery client available.')
+
+    component = resource_node.component
+    resource = resource_node.resource
+
+    yield from filter(
```

Comment:
use generator-comprehension, or - maybe even better - just a plain for-loop
map/filter is considered to be less readable than comprehensions

KISS / readability counts